docs/pvc-recovery.md

```diff
@@ -0,0 +1,279 @@
+
+1. Once all the pods are ready, update your `elasticsearch-recovery` CR to be `Unmanaged`
+
+1. To prevent metrics being scraped for the recovery cluster, delete its `servicemonitor` object
+    - `oc delete servicemonitor -n openshift-logging monitor-elasticsearch-recovery-cluster`
+
+1. Scale down all your recovery nodes
+    - IMPORTANT: Make note of the number of replicas before scaling down your statefulsets (if applicable).
+    - `oc scale deployment --replicas=0 -l cluster-name=elasticsearch-recovery`
+    - `oc scale statefulset --replicas=0 -l cluster-name=elasticsearch-recovery`
+
+1. For each of your deployments|statefulsets in your recovery cluster, `oc edit` and update the following:
+    - the `CLUSTER_NAME` env var must match your prior cluster name, for the sake of these examples it would be `elasticsearch`. If this does not match the prior cluster you will not see your old data when the recovery cluster spins up.
+    - remove the `paused: true` line.
+    - update the `secret` volume definition in `Volumes` to no longer be `elasticsearch-recovery` but instead be `elasticsearch`. If you do not update this the cluster will not be placed because it is waiting for the secret to be available.
+
+1. Get a list of your prior PVCs you wish to recover. We will want to pay attention to the UUIDs for them. Do not include the UUIDs that are being used by the current cluster.
+    - With a PVC name like: `elasticsearch-elasticsearch-cdm-xm9nl5cb-1` the UUID is `xm9nl5cb`.
+    - With a Pod name like: `elasticsearch-cdm-xm9nl5cb-1-6755bb948f-g4plq` the UUID is `xm9nl5cb`.
+
+1. Mount all the previously used PVC for a given UUID to the recovery cluster. for each node:
+    - `oc set volume deployment/<recovery_node_name> --add -t persistentVolumeClaim --claim-name=<name of the old pvc> --name=elasticsearch-storage --overwrite`
+      - Note: while not necessary, it is a good idea to match the PVC numbering to the node numbering for ease of remembering, e.g. `oc set volume deployment/elasticsearch-cdm-ap5obh1j-1 --add -t persistentVolumeClaim --claim-name=elasticsearch-elasticsearch-cdm-xm9nl5cb-1 --name=elasticsearch-storage --overwrite`
+
+1. Scale up all your recovery nodes
+    - `oc scale deployment --replicas=1 -l cluster-name=elasticsearch-recovery`
+    - `oc scale statefulset --replicas=<value before scaledown> -l cluster-name=elasticsearch-recovery`
+
+1. Wait for the recovery cluster to come up and ensure the prior indices exist.
+    - `oc exec -c elasticsearch <es_pod_name> -- indices`
+    - if you do not see your data, verify that you have correctly updated the `CLUSTER_NAME` env var name for each of your nodes.
+
+1. Update the recovery cluster to allow the ES service account to read its indices.
+    - Run this once on the recovery cluster (not for each node)
+    - `oc exec -c elasticsearch <recovery_pod_name> -- sed -i sgconfig/roles_mapping.yml -e "/system.admin/a \    - 'CN=logging-es,OU=OpenShift,O=Logging'"`
```

Comment:
This line still has an issue when copy and pasting. Although the raw code includes four spaces between the backslash and the `- 'CN...` line, the rendered output is trimming three spaces.

Reply:
should be resolved now!

Reply:
LGTM